infrastructure/src/test/java/org/corfudb/infrastructure/batchprocessor/BatchProcessorTest.java

```diff
@@ -0,0 +1,308 @@
+    }
+
+    /**
+     * A helper method that creates a basic message header populated
+     * with default values. Note that the sealEpoch in BatchProcessor
+     * should be equal to the epoch in request header.
+     *
+     * @return   the corresponding HeaderMsg
+     */
+    private HeaderMsg getHeaderHighPriority(boolean ignoreClusterId, boolean ignoreEpoch) {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.HIGH, DEFAULT_SEAL_EPOCH,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), ignoreClusterId, ignoreEpoch);
+    }
+
+
+    /**
+     * A helper method that creates a message header populated with
+     * default values and larger epoch than default header. Note that
+     * this header is only used to construct RESET request, which is just
+     * took for convenience in some exceptional cases.
+     *
+     * @return   the corresponding HeaderMsg
+     */
+    private HeaderMsg getResetHeaderLargerEpoch() {
+        return getHeaderMsg(requestCounter.incrementAndGet(), PriorityLevel.NORMAL, LARGER_SEAL_EPOCH,
+                getUuidMsg(DEFAULT_UUID), getUuidMsg(DEFAULT_UUID), false, true);
+    }
+
+    /**
+     * A helper method that creates a sample LogData object with default values.
+     *
+     * @param address LogData's global address (global tail)
+     * @return        the corresponding HeaderMsg
+     */
+    private LogData getDefaultLogData(long address) {
+        ByteBuf b = Unpooled.buffer();
+        byte[] streamEntry = "Payload".getBytes();
+        Serializers.CORFU.serialize(streamEntry, b);
+        LogData ld = new LogData(DataType.DATA, b);
+        ld.setGlobalAddress(address);
+        ld.setEpoch(0L);
+        return ld;
+    }
+
+    /**
+     * Perform the required preparation before running individual tests.
+     */
+    @Before
+    public void setup() {
+        mockStreamLog = mock(StreamLog.class);
+        batchProcessor = new BatchProcessor(mockStreamLog, DEFAULT_SEAL_EPOCH, true);
+    }
+
+    /**
+     * Test that the BatchProcessor successfully handles a PREFIX_TRIM request.
+     */
+    @Test
+    public void testPrefixTrim() {
+        long epoch = 0L;
+        long sequence = 5L;
+        RequestMsg request = getRequestMsg(getBasicHeader(false, false),
+                getTrimLogRequestMsg(new Token(epoch, sequence)));
+
+        batchProcessor.addTask(BatchWriterOperation.Type.PREFIX_TRIM, request).join();
+        verify(mockStreamLog).prefixTrim(sequence);
+    }
+
+    /**
+     * Test that the BatchProcessor successfully handles a WRITE request.
+     */
+    @Test
+    public void testWrite() {
+        LogData logData = getDefaultLogData(0L);
+        RequestMsg request = getRequestMsg(getBasicHeader(false, false),
+                getWriteLogRequestMsg(logData));
+
+        batchProcessor.addTask(BatchWriterOperation.Type.WRITE, request).join();
+        verify(mockStreamLog).append(0L, logData);
```

Comment:
I don't think these work as expected for `LogData` types, as they are considered equal if the `clientId` and `threadId` are equal.

Reply:
This has been resolved in an offline discussion with @xcchang